packages/template-retail-react-app/app/pages/checkout/partials/shipping-options-list.jsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import React from 'react'
+import {FormattedNumber, useIntl} from 'react-intl'
+import {
+    Box,
+    Flex,
+    Stack,
+    Text,
+    VStack,
+    Radio,
+    RadioGroup
+} from '@salesforce/retail-react-app/app/components/shared/ui'
+import {Controller} from 'react-hook-form'
+import {useShippingMethodsForShipment} from '@salesforce/commerce-sdk-react'
+import PropTypes from 'prop-types'
+import LoadingSpinner from '@salesforce/retail-react-app/app/components/loading-spinner'
+import {STORE_LOCATOR_IS_ENABLED} from '@salesforce/retail-react-app/app/constants'
+
+// Component to handle shipping options for a single shipment (without product cards)
+const ShippingOptionsList = ({shipment, basketId, currency, control}) => {
+    const {formatMessage} = useIntl()
+    const {data: shippingMethods, isLoading: isShippingMethodsLoading} =
+        useShippingMethodsForShipment(
+            {
+                parameters: {
+                    basketId: basketId,
+                    shipmentId: shipment.shipmentId
+                }
+            },
+            {
+                enabled: Boolean(basketId && shipment.shipmentId && shipment.shippingAddress)
+            }
+        )
+
+    if (!shipment.shippingAddress) {
+        return null
+    }
+
+    const fieldName = `shippingMethodId_${shipment.shipmentId}`
+    const defaultValue =
+        shipment.shippingMethod?.id || shippingMethods?.defaultShippingMethodId || ''
+
+    // Filter out pickup shipping methods only if store locator/BOPIS is enabled
+    const applicableShippingMethods = STORE_LOCATOR_IS_ENABLED
```

Comment:
use react-query `select` to return this data instead of doing it in render. this calculation runs on every render.